leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/request/CaliforniumLwM2mRequestSender.java

```diff
@@ -107,8 +105,16 @@ public CaliforniumLwM2mRequestSender(Endpoint secureEndpoint, Endpoint nonSecure
                 destination.canInitiateConnection());
 
         // Handle special observe case
-        if (response != null && response.getClass() == ObserveResponse.class && response.isSuccess()) {
-            observationService.addObservation(destination, ((ObserveResponse) response).getObservation());
+        if (response != null && response.isSuccess()) {
+            Observation observation = null;
+            if (response instanceof ObserveResponse) {
+                observation = ((ObserveResponse) response).getObservation();
+            } else if (response instanceof ObserveCompositeResponse) {
+                observation = ((ObserveCompositeResponse) response).getObservation();
+            }
```

Comment:
Reading this code, I asked myself I we need an interface to avoid this double `if instanceof`. But maybe this is overkill.

Reply:
Maybe we could add interface with generalized ```getObservation()``` and also add specific ```ObserveResponse.getSingleObservation()``` / ```ObserveCompositeResponse.getCompositeObservation()``` methods?

Introducing an interface for Observe*Response could nice logically group these classes together.

And what would be name for this interface? The ObserveResponse would be good one, but is already used :)

Reply:
> And what would be name for this interface? The ObserveResponse would be good one, but is already used :)

I can not find a good name too ...
I agree that a good one would be `ObserveResponse`
But this means that we need to rename (again?) `ObserveResponse` in `SingleObserveResponse` and if we do that (This is a bit strange to keep the `ObserveRequest` name)

I can not find good solution for now :exploding_head:

Reply:
No, I don't want go through renaming Observe* / SingleObserve* again :fearful: 

What do you think about HasObservation interface?

Reply:
> No, I don't want go through renaming Observe* / SingleObserve* again fearful

I can understand :grimacing: 

> What do you think about HasObservation interface?

It sounds a bit strange to me.

Maybe the best move is to keep the code like this finish to implement the feature.
And we will see if we do some more refactoring like this in a second time.